client/doc.go

```diff
@@ -0,0 +1,156 @@
+https://godoc.org/github.com/coniks-sys/coniks-go/bots for the
+CONIKS account validation protocol specification.
+
+- If the registration request is accepted*, the server returns a
+registration proof.
+More specifically, the server returns a proof of absence:
+        reg_pf = (auth_path, str)
+If the server uses the temporary binding protocol extension, reg_pf will
+also include a temporary binding (TB). See
+https://godoc.org/github.com/coniks-sys/coniks-go/protocol/extensions/tb
+for the temporary binding protocol extension specification.
+
+- The client audits the received STR including a hash chain check.
+See https://godoc.org/github.com/coniks-sys/coniks-auditor for the
+auditing protocol specification. *Note: Auditing is not implemented yet.*
+
+- Assuming the audit passes, the client proceeds to verify the TB, if
+included: it verifies the server's signature on th TB and checks that
+they public key in the TB matches the public key it sent to the proxy
+in reg_req.
+
+- Next, the client verifies that auth_path is a proof of absence by
+checking that its private index is invalid for the username, and its
+commitment is invalid for the public key it registered in reg_req.
+The client then recomputes the server's directory root using auth_path,
+and verifies that str includes the recomputed root in its signature.
+
+- If the registration proof is invalid, the client notifies the user. The
+Developing the reporting mechanism is planned for a future release.
+
+* The registration request may be denied for one of the following reasons:
+(1) The client attempts to register a name that already exists in the
+CONIKS key directory (ErrorNameExisted); the server returns a
+privacy-preserving proof of inclusion in its response. (2) The server
+encounters an internal error when attempting to register the name
+(ErrorDirectory).
+
+Assumptions and Possible Attacks: The client accepts the STR in the
+registration proof and assumes that the server has presented it with a
+legitimate history with this initial STR (i.e. Tofu).
+The main downside to the Tofu approach is that this does
+allow a malicious server to place the client
+on a forked branch of its history. However, by auditing the initial STR,
+the client can detect this attack. Alternatively, the client could verify
+the server's entire prior history to ensure that the server has not attempted
+to equivocate about its initial STR. This approach would be more secure,
+but has performance penalties. For now, we believe auditing will mitigate
+this attack, but the need for a full prior history check upon registration
+remains under discussion.
+
+As mentinoed above, for performance reasons, the client does not check
+the server's prior hisotry upon registering a new name. This enables
+a malicious server to remove a legitimate username from the directory
+and allow another user to re-register the name as the registering
+client would not discover the prior existence of the name.
```

Comment:
I'm not sure how comfortable I am with this paragraph.  It might be better framed as the registering client not being able to tell if its name had previously been registered with the server.  At present, it sounds like anyone can register any name they want, which is misleading.